lesson_04/homework04.py

```diff
@@ -0,0 +1,118 @@
+"""This is homework 4."""
+
+
+import logging
+
+# Configure logging
+logging.basicConfig(level=logging.INFO, format='')
+_log = logging.getLogger(__name__)
+adwentures_of_tom_sawer = """\
+Tom gave up the brush with reluctance in his .... face but alacrity
+in his heart. And while
+the late steamer
+"Big Missouri" worked ....
+and sweated
+in the sun,
+the retired artist sat on a barrel in the .... shade close by,
+dangled his legs,
+munched his apple, and planned the slaughter of more innocents.
+There was no lack of material;
+boys happened along every little while;
+they came to jeer, but .... remained to whitewash. ....
+By the time Ben was fagged out,
+Tom had traded the next chance to Billy Fisher for
+a kite, in good repair;
+and when he played
+out, Johnny Miller bought
+in for a dead rat and a string to swing it with—and so on, and so on,
+hour after hour. And when the middle of the afternoon came, from being a
+poor poverty, stricken boy in the .... morning, Tom was literally
+rolling in wealth."""
+
+# ПЕРЕЗАПИСУЙТЕ зміст змінної adwentures_of_tom_sawer у завданнях 1-3
+# task 01 ==
+""" Дані у строці adwentures_of_tom_sawer
+розбиті випадковим чином, через помилку.
+треба замінити кінець абзацу на пробіл .replace("\n", " ")"""
+
+_log.info(adwentures_of_tom_sawer)
+str_result = adwentures_of_tom_sawer.replace('\n', ' ')
+_log.info(str_result)
+_log.info('=' * 75)
+
+# task 02 ==
+""" Замініть .... на пробіл
+"""
+
+str_result2 = str_result.replace(' .... ' and ' ....', ' ')

```

Comment:
Better to avoid such constructions - just use two replace operations if needed
Best practice - one replace at the time.